File: Python3/Dijkstra.py
import queue
MAX = 100
INF = int(1e9)
class Node :
    def __init__(self, id, dist) :
        self.dist = dist
        self.id = id
    def __lt__(self, other) :
        return self.dist <= other.dist

def Dijkstra(s, graph) :
    pq = queue.PriorityQueue()
    pq.put(Node(s, 0))
    dist[s] = 0
    while pq.empty() == False :
        top = pq.get()
        node = top.id
        d = top.dist
        for neighbor in graph[node] :
            if d + neighbor.dist < dist[neighbor.id] :
                dist[neighbor.id] = d + neighbor.dist
                pq.put(Node(neighbor.id, dist[neighbor.id]))
                path[neighbor.id] = node

if __name__ == '__main__' :
    n = map(int, input().split())
    s = 0, t = 4
    graph = [[] for i in range(n + 5)]
    dist = [INF for i in range(n + 5)]
    path = [-1 for i in range(n + 5)]
    for i in range(n) :
        d = list(map(int, input().split()))
        for j in range(n) :
            if d[j] > 0:
                graph[i].append(Node(j, d[j]))
    Dijkstra(s, graph)
    ans = dist[t]
    print(ans)

File: C++/Dijkstra.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>
#include <queue>
#include <functional>
using namespace std;
#define MAX 100
const int INF = 1e9;
vector<vector<pair<int, int> > > graph;
vector<int> dist(MAX, INF);
int path[MAX];
void Dijkstra(int s, vector<vector<pair<int, int> > > &graph){
    priority_queue<pair<int, int>, vector<pair<int, int> >,
        greater<pair<int, int> > > pq;
    pq.push(pair<int, int>(0, s));
    dist[s] = 0;
    while (!pq.empty()) {
        pair<int, int> top = pq.top();
        pq.pop();
        int node = top.second;
        int d = top.first;
        for (int i = 0; i < graph[node].size(); ++i) {
            pair<int, int> neighbor = graph[node][i];
            if (d + neighbor.second < dist[neighbor.first]) {
                dist[neighbor.first] = d + neighbor.second;
                pq.push(pair<int, int>(dist[neighbor.first], neighbor.first));
                path[neighbor.first] = node;
            }
        }
    }
}
int main() {
    int n, s, t;
    dist = vector<int>(MAX, INF);
    cin >> n;
    s = 0, t = 4;
    graph = vector<vector<pair<int, int> > >(MAX + 5, vector<pair<int, int> >());
    int d = 0;
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++) {
            cin >> d;
            if(d > 0)
                graph[i].push_back(pair<int, int>(j, d));
        }
    Dijkstra(s, graph);
    int ans = dist[t];
    cout << ans;
    return 0;
}
